src/v/cluster/members_manager.h

```diff
@@ -190,7 +230,11 @@ class members_manager {
     simple_time_jitter<model::timeout_clock> _join_retry_jitter;
     const std::chrono::milliseconds _join_timeout;
     const consensus_ptr _raft0;
+
+    ss::sharded<controller_stm>& _controller_stm;
```

Comment:
woulld it be possible not to depend on the `_controller_stm` here ? It creates circular dependency that will be hard to untangle.

Reply:
Current thought is to maybe have this depend on `members_frontend`, which doesn't feel great because the call we need (replicate on the controller log right now) is a bit different from the current frontend pattern (dispatch to leader, potentially replicating to the controller log).

I left this as is for now, but may revisit if you feel strongly about it / it causes trouble.